### PR TITLE
Group Tasks

### PR DESCRIPTION
**Pull request by RitobrotoMukherjee**
### Group task
- Reference [UML Diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png)
- Create Item class in a separate .rb file.
- All Item class properties visible in the diagram should be defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created._
- Add all methods visible in the diagram.
- Implement methods:
    - can_be_archived?() in the Item class
        - should return true if published_date is older than 10 years
        - otherwise, it should return false
    - move_to_archive() in the Item class
        - should reuse can_be_archived?() method
        - should change the archived property to true if the result of the can_be_archived?() method is true
        - should do nothing if the result of the can_be_archived?() method is false
 - Create a main.rb file that will serve as your console app entry-point.
 - Implement startup actions:
    - Present the user with a list of options to perform.
    - Let users choose an option.
    - If needed, ask for parameters for the option.
    - Have a way to quit the app.

### CODE REVIEW

**Review by aliatayee — APPROVED** [review]

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-ruby@v1
+        with:
+          ruby-version: 3.1.x
+      - name: Setup RSpec
+        run: |
+          [ -f Gemfile ] && bundle
+          gem install --no-document rspec -v '>=3.0, < 4.0'
+      - name: RSpec Report
```
Comment (ksigei) [inline .github/workflows/tests.yml:18]: :heavy_check_mark: Workflow

File: .rubocop.yml [hunk 2]
```diff
@@ -50,10 +50,3 @@ Style/HashTransformKeys:
   Enabled: false
 Style/HashTransformValues:
   Enabled: false
-Layout/EndOfLine:
-  Enabled: False
-Metrics/BlockLength:
-  Exclude:
-    - 'Rakefile'
-    - '**/*.rake'
-    - 'spec/**/*.rb'
```
Comment (ksigei) [inline .rubocop.yml:59]: :heavy_check_mark: Tests

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    app.perform_main_operation(input)
+  end
+
+  app.save_data
+  puts "\nThank You For Using Catalog App!"
+end
+
+app = App::INIT.new
+
+main(app)
```
Comment (ksigei) [inline main.rb:22]: :heavy_check_mark: App entry point

**Review by aliatayee — APPROVED** [review]
Good job guys!
## STATUS APPROVED!
You did a great job. from my side this task is complete we can merge it.

File: models/item.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  private
+
+  def can_be_archived?
+    date_diff = Date.today.year - Date.parse(@published_date).year
+    return false if date_diff.to_i > 10
+
+    true
+  end
+end
```
Comment (ksigei) [inline models/item.rb:48]: :heavy_check_mark: Item class

**Review by ksigei — APPROVED** [review]
Hi @RitobrotoMukherjee , @aliatayee , @bigmosi  :wave: 
Congratulations on the work done, we are going on well team :100: 
The task is complete, nothing else remaining other than to merge it and proceed to the next task
:smile:

**Review by bigmosi — APPROVED** [review]
@ksigei nice code and everything is complete

**Comment by bigmosi** [comment]
We can merge it

**Pull request merged**